### PR TITLE
Update persistence-snapshot.md

### PR DESCRIPTION
**Pull request by sebastian-alfers**
Add more clarification around deleting snaphots.

### CODE REVIEW

File: akka-docs/src/main/paradox/typed/persistence-snapshot.md [hunk 1]
```diff
@@ -95,7 +95,7 @@ Scala
 Java
 :  @@snip [BasicPersistentBehaviorTest.java](/akka-persistence-typed/src/test/java/jdocs/akka/persistence/typed/BasicPersistentBehaviorTest.java) { #retentionCriteria #snapshottingPredicate }
 
-Snapshot deletion is triggered after saving a new snapshot.
+Snapshot deletion is triggered after successfully saving a new snapshot. Is saving a snaphot failed, snapshot deletion is not triggered.
```
Comment (johanandren) [inline akka-docs/src/main/paradox/typed/persistence-snapshot.md:98]:
```suggestion
Snapshot deletion is triggered after successfully saving a new snapshot. If saving a snapshot failed, snapshot deletion is not triggered.
```

**Comment by leviramsey** [comment]
I forget how clear the docs are on elective (i.e. not triggered by retention criteria) snapshots not triggering deletion, but I know it wasn't that long ago that I was surprised to learn that.

File: akka-docs/src/main/paradox/typed/persistence-snapshot.md (hunk 1 above)
Comment (octonato) [reply akka-docs/src/main/paradox/typed/persistence-snapshot.md]: I think it should says: "If saving a snapshot fails, the snapshot deletion is not triggered"

**Review by patriknw — APPROVED** [review]
LGTM

**Pull request merged**